### DISCUSSION TITLE
allowedVersions doesn't use normalized version

### DISCUSSION THREAD

**Discussion by satanTime** [renovatebot/renovate#25708] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I want to apply `allowedVersions` to `nvm`:

```json
    {
      "matchPackageNames": [
        "satantime/puppeteer-node",
        "@types/node",
        "nodejs/node"
      ],
      "allowedVersions": "/^\\d*[02468]\\./",
      "branchTopic": "node-{{{newMajor}}}{{#if separateMinorPatch}}{{#if isPatch}}.{{{newMinor}}}{{/if}}{{/if}}.x{{#if isLockfileUpdate}}-lockfile{{/if}}"
    }
```

However, all PRs don't have `.nvmrc` at all, despite `nodejs/node`.
Job logs show:

```json
      {
        "deps": [
          {
            "depName": "node",
            "currentValue": "18.18.2",
            "datasource": "github-tags",
            "packageName": "nodejs/node",
            "updates": [],
            "versioning": "node",
            "warnings": [],
            "sourceUrl": "https://github.com/nodejs/node",
            "registryUrl": "https://github.com",
            "currentVersion": "18.18.2",
            "isSingleVersion": true,
            "fixedVersion": "18.18.2"
          }
        ],
        "packageFile": ".nvmrc"
      },
```

It looks obvious that based on `currentValue`, `currentVersion`, `fixedVersion` it should be updated to `20.x.x`, but nothing happens.

The issue here is that `nvm` version includes `v`, and the right `allowedVersions` is `^v?\\d*[02468]\\./"`, but it's impossible to guess about this.

I would propose to apply `allowedVersions` to the parsed (extracted) version, so I wouldn't worry whether `v` should be or shouldn't be here.

### Relevant debug logs

<details><summary>Logs</summary>

```
      {
        "deps": [
          {
            "depName": "node",
            "currentValue": "18.18.2",
            "datasource": "github-tags",
            "packageName": "nodejs/node",
            "updates": [],
            "versioning": "node",
            "warnings": [],
            "sourceUrl": "https://github.com/nodejs/node",
            "registryUrl": "https://github.com",
            "currentVersion": "18.18.2",
            "isSingleVersion": true,
            "fixedVersion": "18.18.2"
          }
        ],
        "packageFile": ".nvmrc"
      },
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
It's behaving as designed. If you want config help for how to work around it, or to request a feature, create the appropriate discussion

**Reply by satanTime** [reply, marked as answer]
okay - if it's by design - how should I understand that `v` is missing? logs don't show any field with `v` in versions.

**Reply by rarkins** [reply]
Please raise a config help discussion
